megamek/src/megamek/client/ui/swing/util/UIUtil.java

```diff
@@ -409,7 +411,169 @@ public static void scaleMenu(final JComponent popup) {
+    }
+
+    /**
+     *
+     * @param imgSplashFile path to an image on disk
+     * @param parent component
+     * @return a JLabel setup to the correct size to act as a splash screen
+     */
+    public static JLabel createSplashComponent(String imgSplashFile, Component parent) {
+        // Use the current monitor so we don't "overflow" computers whose primary
+        // displays aren't as large as their secondary displays.
+        Dimension scaledMonitorSize = getScaledScreenSize( parent.getGraphicsConfiguration().getDevice().getDisplayMode());
+
+        Image imgSplash = parent.getToolkit().getImage(imgSplashFile);
+
+        // wait for splash image to load completely
+        MediaTracker tracker = new MediaTracker(parent);
+        tracker.addImage(imgSplash, 0);
+        try {
+            tracker.waitForID(0);
+        } catch (InterruptedException ignored) {
+            // really should never come here
+        }
+
+        return createSplashComponent(imgSplash, parent, scaledMonitorSize);
+    }
+
+    /**
+     *
+     * @param imgSplash an image
+     * @param observer
+     * @param scaledMonitorSize the dimensions of the monitor taking into account display scaling
+     * @return a JLabel setup to the correct size to act as a splash screen
+     */
+    public static JLabel createSplashComponent(Image imgSplash, ImageObserver observer, Dimension scaledMonitorSize) {
+        JLabel splash;
+        Dimension maxSize = new Dimension(
+                (int) (scaledMonitorSize.width*0.75),
+                (int) (scaledMonitorSize.height * 0.75));
+
+        if (imgSplash != null) {
+            imgSplash = UIUtil.constrainImageSize(imgSplash, null, maxSize.width, maxSize.height );
+            Icon icon = new ImageIcon(imgSplash);
+            splash = new JLabel(icon);
+        } else {
+            splash = new JLabel();
+        }
+
+        Dimension splashDim = new Dimension(
+                imgSplash == null ? maxSize.width : imgSplash.getWidth(observer),
+                imgSplash == null ? maxSize.height : imgSplash.getHeight(observer));
+
+        splash.setMaximumSize(splashDim);
+        splash.setMinimumSize(splashDim);
+        splash.setPreferredSize(splashDim);
+
+        return splash;
+    }
+
+
+    public static void keepOnScreen(JFrame component) {
+
+        DisplayMode currentMonitor = component.getGraphicsConfiguration().getDevice().getDisplayMode();
+        Dimension scaledScreenSize = UIUtil.getScaledScreenSize(currentMonitor);
+
+        Point pos = component.getLocationOnScreen();
+        Dimension size = component.getSize();
+        Rectangle r = new Rectangle(scaledScreenSize);
+
+        // center and size if out of bounds
+        if ( (pos.x < 0) || (pos.y < 0) ||
+                (pos.x + size.width > scaledScreenSize.width) ||
+                (pos.y + size.height > scaledScreenSize.getHeight())) {
+            component.setLocationRelativeTo(null);
+        }
+    }

```

Comment:
Seeing that this is quite a bit of code and has no direct connection to the other stuff in UIUtil I'm tempted to suggest moving this to its own "MonitorUtil" or something like that. But I know that Windchild suggested UIUtil and there's nothing wrong with that either. I'll leave it to you. We probably could use some more utility methods for handling multi monitor setups as we often have problems with these.

Reply:
I'll leave it here for now.